clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -180,4 +213,13 @@ public Node leastLoadedNode(long now) {
         return this.node;
     }
 
+    /**
+     * Request predicates give the tester the ability to set an expectation on the request that
```

Comment:
This comment is not very intuitive to understand: how about: ".. set an expectation on the request that this response is going to be returned for..."